src/integration-test/java/com/commercetools/sync/integration/commons/utils/ITUtils.java

```diff
@@ -42,44 +39,50 @@ public static void deleteTypesFromTargetAndSource() {
     }
 
     /**
-     * Builds a JSON String that represents the fields of the supplied instance of {@link BaseSyncStatistics}.
-     * Note: The order of the fields in the built JSON String depends on the order of the instance variables in this
-     * class.
+     * Applies the {@code resourceToRequestMapper} function on each page, fetched from the supplied
+     * {@code queryRequestSupplier} on the supplied {@code ctpClient}, to map each resource to a {@link SphereRequest}
+     * and then executes these requests in parallel within each page.
      *
-     * @param statistics the instance of {@link BaseSyncStatistics} from which to create a JSON String.
-     * @return a JSON String representation of the statistics object.
+     * @param ctpClient               defines the CTP project to apply the query on.
+     * @param queryRequestSupplier    defines a supplier which, when executed, returns the query that should be made on
+     *                                the CTP project.
+     * @param resourceToRequestMapper defines a mapper function that should be applied on each resource, in the fetched
+     *                                page from the query on the specified CTP project, to map it to a
+     *                                {@link SphereRequest}.
      */
-    public static String getStatisticsAsJSONString(@Nonnull final BaseSyncStatistics statistics)
-        throws JsonProcessingException {
-        final ObjectMapper mapper = new ObjectMapper();
-        return mapper.writeValueAsString(statistics);
+    public static <T extends Resource, C extends QueryDsl<T, C>> void queryAndExecute(
+        @Nonnull final SphereClient ctpClient,
+        @Nonnull final Supplier<QueryDsl<T, C>> queryRequestSupplier,
+        @Nonnull final Function<T, SphereRequest<T>> resourceToRequestMapper) {
```

Comment:
Is it a `resourceToRequestMapper` or rather `responseConsumer`?

Reply:
@butenkor I think it's more `resourceToRequestMapper` as it represents a function that maps/converts a resource of type `T` to a `SphereRequest`

Reply:
Then maybe `responseToRequestMapper`? For me it seems not obvious what `resource` is and where does it come from.

Reply:
but response is more representing the entire response from CTP rather than an actual "resource" object coming in the page of the response. It is also described/explained in the [Javadoc](https://github.com/commercetools/commercetools-sync-java/pull/241/files/85321d1b9da1ddc40eefcce2957925a12797a208#diff-1d8a8cea016d4c60e4d2fba61d10445cR49) :)

Reply:
the response of a query request is a page of resources.